### PR TITLE
Fetch build attributes via board type not MbedTarget

### PR DESCRIPTION
**Pull request by rosiemiller2**
### Description
This removes the need to create a full MbedTarget object in order to get build attributes. Just need to know the build type.

### Test Coverage

- [x]  This change is covered by existing or additional automated tests.
- [ ]  Manual testing has been performed (and evidence provided) as automated testing was not feasible.
- [ ]  Additional tests are not required for this change (e.g. documentation update).

### CODE REVIEW

File: mbed_targets/mbed_target.py [hunk 1]
```diff
@@ -76,13 +76,13 @@ def from_offline_target_entry(cls, target_entry: dict) -> "MbedTarget":
         )
 
 
-def get_target_build_attributes(mbed_target: MbedTarget, path_to_targets_json: str) -> Any:
+def get_target_build_attributes(mbed_target_board_type: str, path_to_targets_json: str) -> Any:
```
Comment (muchzill4) [inline mbed_targets/mbed_target.py]: Not part of this pr, but do you think it's worth exposing this as public api for this package? I don't know what the long term plans are.

Comment (rosiemiller2) [reply mbed_targets/mbed_target.py]: I think it does need to be part of the public API of the package since I need to be able to at least get the build attributes somehow from `mbed-build` but I am happy for suggestions about how best to do that.

Comment (muchzill4) [reply mbed_targets/mbed_target.py]: I'd say expose it in `__init__.py` of the package: https://github.com/ARMmbed/mbed-targets/blob/master/mbed_targets/__init__.py#L20 . 🤔

Comment (rosiemiller2) [reply mbed_targets/mbed_target.py]: I am tempted to create a new file `mbed_targets_build_attributes.py` (or something similar) where I mirror what is going on in `mbed_targets.py` a bit, create a `MbedTargetBuildAttributes` object and move over the `get_target_build_attributes` function.

Comment (rosiemiller2) [reply mbed_targets/mbed_target.py]: Oooh! Sorry I was completely misunderstanding. Absolutely 👍

Comment (rosiemiller2) [reply mbed_targets/mbed_target.py]: I definitely will, but I might do it after I have made the changes described above if thats ok? Maybe this PR is a bit irrelevant...

Comment (rosiemiller2) [reply mbed_targets/mbed_target.py]: Yeah, sorry about this. I'll close this one and do it all in one go. Thanks for the pointer to exposing in the init file.
